lib/abrp.js

```diff
@@ -176,14 +178,29 @@ function mapMetricsToTelemetry(metrics) {
   // Array.prototype.includes() not supported in duktape
   // TODO: confirm if is_charging is supposed to be true if regenerative braking is
   // charging the battery.
-  const is_charging = chargingStates.indexOf(metrics['v.c.state']) > -1
+  var is_charging = chargingStates.indexOf(metrics['v.c.state']) > -1;
   // The instrument range reported by the Nissan Leaf OVMS metrics doesn't
   // appear to update when the car is parked and then charged. So, use the
   // generic vehicle ideal range when it's more than 10% of the reported
   // instrument range.
   const instrumentRange = round(metrics['xnl.v.b.range.instrument']) || 0
   const idealRange = round(metrics['v.b.range.ideal'])
   // https://documenter.getpostman.com/view/7396339/SWTK5a8w
+
+  // Add switches based on v.type
+  const vehicleType = metrics['v.type'];
+  switch (vehicleType) {
+    case 'SUBSOL':
+    case 'TOYBZ4X':
+      is_charging = metrics['v.c.charging'];
```

Comment:
If this is the correct metric to use for those vehicle types this will need to be converted to boolean as docs suggest this is a string with values on or off.

Reply:
> If this is the correct metric to use for those vehicle types this will need to be converted to boolean as docs suggest this is a string with values no or yes.

From the OVMS source code, this metrics is a boolean.
#define MS_V_CHARGE_INPROGRESS      "v.c.charging"
OvmsMetricBool*   ms_v_charge_inprogress;             // True = currently charging